src/pages/categories/Categories.jsx

```diff
@@ -1,7 +1,153 @@
 import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import { Box, Typography, Paper, InputBase, Autocomplete } from '@mui/material'
+import SearchIcon from '@mui/icons-material/Search'
+import { useTranslation } from 'react-i18next'
+import { styles } from '~/pages/categories/Categories.styles.js'
+import ArrowForwardIcon from '@mui/icons-material/ArrowForward'
+import AppButton from '~/components/app-button/AppButton'
+import HashLink from '~/components/hash-link/HashLink'
+import { categoryService } from '~/services/category-service'
+import { useEffect, useState, useMemo } from 'react'
+import { authRoutes } from '~/router/constants/authRoutes'
+import CategoryItemCard from '~/components/category-item-card/CategoryItemCard'
 
 const Categories = () => {
-  return <PageWrapper>Categories</PageWrapper>
+  const [categoriesData, setCategoriesData] = useState([])
+  const [inputValue, setInputValue] = useState('')
+  const [selectedCategory, setSelectedCategory] = useState(null)
+  const [actualSearch, setActualSearch] = useState(null)
+
+  const { t } = useTranslation()
+
+  const { categories, subjects, findOffers } = authRoutes
+
+  const searchedCategories = useMemo(() => {
+    if (!actualSearch) return categoriesData
+
+    const normilizedSearch = normilizeString(actualSearch)
+
+    return categoriesData.filter((category) =>
+      normilizeString(category.name).includes(normilizedSearch)
+    )
+  }, [categoriesData, actualSearch])
+
+  useEffect(() => {
+    const fetchCategories = async () => {
+      try {
+        const response = await categoryService.getCategories()
+        setCategoriesData(response.data.items)
+      } catch (error) {
+        console.error('Error fetching categories:', error)
+      }
+    }
+
+    fetchCategories()
+  }, [])
+
+  const handleCategorySearch = () => {
+    setActualSearch(inputValue)
+  }
+
+  const handleCategoryChange = (event, value) => {
+    setSelectedCategory(value)
+    setInputValue(value?.name || '')
+    setActualSearch(value?.name)
+  }
+
+  function normilizeString(v) {
+    return v?.trim().toLowerCase()
+  }
```

Comment:
We have other autocompletes, is this feature useful for them?

Reply:
Why do you think we need a "Search" button with the current implementation?
- it just won't search for words starting with lowercase and spaces without this function. 

We have other autocompletes, is this feature useful for them?
- the feature will be useful for other autocompletes.  In other words, are you telling me it would be better to move the function to a separate file and then export it?